interp/constrextern.ml

```diff
@@ -274,17 +274,8 @@ let pattern_printable_in_both_syntax (ind,_ as c) =
 
  (* Better to use extern_glob_constr composed with injection/retraction ?? *)
 let rec extern_cases_pattern_in_scope (scopes:local_scopes) vars pat =
-  (* pboutill: There are letins in pat which is incompatible with notations and
-     not explicit application. *)
-  match pat with
-    | PatCstr(loc,cstrsp,args,na)
-	when !Flags.in_debugger||Inductiveops.constructor_has_local_defs cstrsp ->
-      let c = extern_reference loc Id.Set.empty (ConstructRef cstrsp) in
-      let args = List.map (extern_cases_pattern_in_scope scopes vars) args in
-      CPatCstr (loc, c, Some (add_patt_for_params (fst cstrsp) args), [])
-    | _ ->
   try
-    if !Flags.raw_print || !print_no_symbol then raise No_match;
+    if !Flags.in_debugger || !Flags.raw_print || !print_no_symbol then raise No_match;
```

Comment:
The flag `in_debugger` is set by the debugger printers. The debugger is not running in the same space as the `coqtop.byte` being debugged and it has no access to the global environment of the executable being run. So, e.g., there is no nametab available to revert internal names into user-friendly names, etc. The role of the `in_debugger` flag is to inform the "constrexterning" system that it should not be a fatal failure when looking within such table fails, providing instead a fallback, even if a visually ugly one (e.g. `_UNBOUND_REL_1`).

This is related to why the debugger printers sometimes fail. These are generally attempts to access some "constexterning" data unavailable without a proper handler to catch the exception.

Hope I'm clear enough.